lib/ExpensiMark.d.ts

```diff
@@ -0,0 +1,118 @@
+declare type Replacement = string | ((...args: string[]) => string);
+declare type Rule = {
+    name: string;
```

Comment:
How about making this a union of literal strings? We can collect the set of possible rule names from `ExpensiMark.js`

Reply:
I don't know if rules are accessed from client codes of the `ExpensiMark` class. But if so, it might be a good idea to specify what kind of string values `name` can have.

For creating union types for `Rule` as suggested by @neil-marcellini, I don't know if we need to be that specific.

Reply:
Rules aren't accessed in the App, so I don't think we need to type them as suggested by @neil-marcellini. Changed to union of literal strings 👌